File: frontend_flutter/lib/models/notification.dart
class Notification {
  final String id;
  final String userId;
  final String title;
  final String message;
  final String notificationType;
  final bool isRead;
  final String? link;
  final String createdAt;

  Notification({
    required this.id,
    required this.userId,
    required this.title,
    required this.message,
    required this.notificationType,
    required this.isRead,
    this.link,
    required this.createdAt,
  });

  factory Notification.fromJson(Map<String, dynamic> json) {
    return Notification(
      id: json['id'],
      userId: json['user_id'],
      title: json['title'],
      message: json['message'],
      notificationType: json['notification_type'],
      isRead: json['is_read'],
      link: json['link'],
      createdAt: json['created_at'],
    );
  }
}

File: backend/app/models/notification.py
from sqlalchemy import Column, String, Text, Boolean, ForeignKey, Enum
from sqlalchemy.sql import func
from ..database import Base
from datetime import datetime
import enum

class NotificationType(str, enum.Enum):
    INFO = "info"
    SUCCESS = "success"
    WARNING = "warning"
    ERROR = "error"
    ASSIGNMENT = "assignment"
    GRADE = "grade"

class Notification(Base):
    __tablename__ = "notifications"
    
    id = Column(String(36), primary_key=True)
    user_id = Column(String(36), ForeignKey("users.id", ondelete="CASCADE"), nullable=False, index=True)
    title = Column(String(255), nullable=False)
    message = Column(Text, nullable=False)
    notification_type = Column(Enum(NotificationType), default=NotificationType.INFO)
    is_read = Column(Boolean, default=False)
    link = Column(String(500))
    created_at = Column(String(50), default=lambda: datetime.utcnow().isoformat())
